src/main/java/de/tum/in/www1/artemis/service/quiz/QuizScheduleService.java

```diff
@@ -0,0 +1,220 @@
+            result.setRated(true);
+            result.setAssessmentType(AssessmentType.AUTOMATIC);
+            result.setCompletionDate(quizSubmission.getSubmissionDate());
+            result.setSubmission(quizSubmission);
+
+            quizSubmission.calculateAndUpdateScores(quizExercise.getQuizQuestions());
+            result.evaluateQuizSubmission();
+
+            quizSubmissionRepository.save(quizSubmission);
+            resultRepository.save(result);
+        });
+        quizStatisticService.recalculateStatistics(quizExercise);
+    }
+
+    /**
+     * cancels the quiz start for the given exercise id, e.g. because the quiz was deleted or the quiz start date was changed
+     *
+     * @param quizExerciseId the quiz exercise for which the quiz start should be canceled
+     */
+    public void cancelScheduledQuizStart(Long quizExerciseId) {
+        scheduleService.cancelScheduledTaskForLifecycle(quizExerciseId, ExerciseLifecycle.START);
+    }
+
+    /**
+     * Internal method to start and send the {@link QuizExercise} to the clients when called
+     */
+    void executeQuizStartNowTask(Long quizExerciseId) {
+        log.debug("Sending quiz {} start", quizExerciseId);
+        QuizExercise quizExercise = quizExerciseRepository.findByIdWithQuestionsAndStatisticsElseThrow(quizExerciseId);
+
+        if (quizExercise.getQuizMode() != QuizMode.SYNCHRONIZED) {
+            throw new IllegalStateException();
+        }
+
+        // TODO: quiz cleanup: We create a batch that has just started here because we can't access QuizBatchService here because of dependencies
+        var quizBatch = new QuizBatch();
+        quizBatch.setQuizExercise(quizExercise);
+        quizBatch.setStartTime(ZonedDateTime.now());
+        quizExercise.setQuizBatches(Set.of(quizBatch));
+
+        quizBatchRepository.save(quizBatch);
+
+        SecurityUtils.setAuthorizationObject();
+        quizMessagingService.sendQuizExerciseToSubscribedClients(quizExercise, quizBatch, START_NOW);
+    }
+
+    @EventListener(ApplicationReadyEvent.class)
+    public void applicationReady() {
+        // schedule the task after the application has started to avoid delaying the start of the application
+        scheduler.schedule(this::scheduleRunningExercisesOnStartup, Instant.now().plusSeconds(QUIZ_EXERCISE_SCHEDULE_DELAY_SEC));
+    }
+
+    /**
+     * Schedule all quiz exercises that will start in the future on application startup
+     */
+    public void scheduleRunningExercisesOnStartup() {
+        try {
+            Collection<String> activeProfiles = Arrays.asList(env.getActiveProfiles());
+            if (activeProfiles.contains(JHipsterConstants.SPRING_PROFILE_DEVELOPMENT)) {
+                // only execute this on production server, i.e. when the prod profile is active
+                // NOTE: if you want to test this locally, please comment it out, but do not commit the changes
+                return;
```

Comment:
Is there a good reason we disable this for dev environments? I don't really see the benefit of disabling it and it could be confusing when testing.